src/lazy/expanded/compiler.rs

```diff
@@ -50,6 +54,17 @@ impl ExpansionAnalysis {
         }
     }
 
+    /// Returns an expansion analysis for a macro that will produce a single value that may or may
+    /// not be a system value.
+    pub const fn possible_system_value() -> Self {
```

Comment:
🪧 I added a few more named `ExpansionAnalysis` constructors so their intent can be expressed at a high level.